### PR TITLE
Sf 1110 - Add label overwrites for pastPurchase sorts in the `tags` section in the config

### PR DESCRIPTION
**Pull request by zoomchik**

### CODE REVIEW

File: src/sort/index.ts [hunk 1]
```diff
@@ -61,15 +62,20 @@ class Sort {
     }
   }
 
-  getPastPurchasesLabel(sort: Store.Sort) {
-    return sort.field;
+  getPastPurchasesLabel(sort: Store.Sort, index: number) {
```
Comment (Vic-Dev) [inline src/sort/index.ts]: this and `getLabel` use the same logic, so move that logic into a function which they both call

**Review by Vic-Dev — CHANGES_REQUESTED** [review]
One change

File: src/sort/index.ts [hunk 2]
```diff
@@ -61,15 +62,20 @@ class Sort {
...
+      return `${sort.field} ${sort.descending ? 'Descending' : 'Ascending'}`;
+    }
   }
 }
 
 interface Sort extends Tag<Sort.Props, Sort.State> {}
 namespace Sort {
   export interface Props extends Tag.Props {
     labels: string[];
+    pastPurchasesLabels: string[];
```
Comment (Vic-Dev) [inline src/sort/index.ts]: Just noticing that the prop definition doesn't include `storeSection` please add it

Comment (Vic-Dev) [inline src/sort/index.ts]: Just noticing that the prop definition doesn't include `storeSection` please add it

File: src/sort/index.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import { configurable, provide, tag, Events, Selectors, Store, StoreSections, Ta
 class Sort {
   props: Sort.Props = {
     labels: [],
+    pastPurchasesLabels: []
```
Comment (Vic-Dev) [inline src/sort/index.ts]: should have a default for `storeSection` prop as `StoreSections.SEARCH`

**Review by Vic-Dev — CHANGES_REQUESTED** [review]
A couple additional things not directly related to this ticket, but important to add. Thank you!

File: src/sort/index.ts (hunk 2 above)
Comment (Vic-Dev) [reply src/sort/index.ts]: Actually it's defined on `Tag`. I added a commit to undo this work

File: src/sort/index.ts [hunk 4]
```diff
@@ -41,35 +42,32 @@ class Sort {
...
           selected: sorts.selected === index,
         }));
     }
   }
 
-  getLabel(sort: Store.Sort, index: number) {
-    if (index < this.props.labels.length) {
+  getLabel(sort: Store.Sort, index: number, labels: string[]) {
+    if (index < labels.length) {
       return this.props.labels[index];
```
Comment (jmbtutor) [inline src/sort/index.ts]: You need to differentiate between search sort labels and past purchase sort labels. This here always takes the search sort labels.

File: src/sort/index.ts [hunk 5]
```diff
@@ -41,35 +42,32 @@ class Sort {
...
-  getPastPurchasesLabel(sort: Store.Sort) {
-    return sort.field;
-  }
 }
 
 interface Sort extends Tag<Sort.Props, Sort.State> {}
 namespace Sort {
   export interface Props extends Tag.Props {
     labels: string[];
+    pastPurchasesLabels: string[];
```
Comment (jmbtutor) [inline src/sort/index.ts]: These props should be optional.

**Review by jmbtutor — DISMISSED** [review]
There's no differentiation between search and past purchase labels in `getLabel`. Please fix that and related tests. As well, please rebase on to the latest `master`.

File: src/sort/index.ts (hunk 4 above)
Comment (zoomchik) [reply src/sort/index.ts]: My fault, fixed

**Review by Vic-Dev — DISMISSED** [review]

**Review by Vic-Dev — APPROVED** [review]

**Pull request merged**